bundles/mapping/mapwfs2/service/WFSLayerService.js

```diff
@@ -139,35 +139,23 @@ Oskari.clazz.define(
          * Handles status of selected features
          */
         setWFSFeaturesSelections: function (layerId, featureIds, makeNewSelection) {
-            var me = this;
-
+            let updatedIds;
             if (makeNewSelection) {
-                this.__removeFeatureSelectionForLayer(layerId);
-                me.getWFSSelections().push({ 'layerId': layerId, 'featureIds': featureIds });
+                updatedIds = featureIds;
             } else {
-                const existingFeatureSelections = this.getSelectedFeatureIds(layerId);
-                // no existing selections -> add all
-                if (!existingFeatureSelections || existingFeatureSelections.length === 0) {
-                    existingFeatureSelections.push(featureIds);
+                const existingIds = this.getSelectedFeatureIds(layerId);
+                // Either add all featureIds or remove all feature Ids from selection. Don't mix.
+                const someSelected = existingIds.some(selected => featureIds.includes(selected));
```

Comment:
How do you feel about renaming these for clarity?
- `updatedIds` -> `selectedFeatureIds`
- `existingIds` -> `previousSelectedFeatureIds`
- `someSelected` -> `shouldRemoveFeaturesFromSelection`